module/validation/receipt_validator.go

```diff
@@ -151,6 +155,22 @@ func (v *receiptValidator) subgraphCheck(result *flow.ExecutionResult) error {
 	return nil
 }
 
+func (v *receiptValidator) resultChainCheck(result *flow.ExecutionResult, prevResult *flow.ExecutionResult) error {
+	//  enforce that execution results form chain
```

Comment:
Shall we turn the comment into a goDoc (?)
```suggestion
// resultChainCheck enforces that the end state of the parent result
// matches the current result's start state
func (v *receiptValidator) resultChainCheck(result *flow.ExecutionResult, prevResult *flow.ExecutionResult) error {
```